clients/src/main/java/org/apache/kafka/common/utils/Utils.java

```diff
@@ -331,6 +331,41 @@ public static ByteBuffer wrapNullable(byte[] array) {
         return Arrays.copyOf(src, src.length);
     }
 
+    /**
+     * Compares two character arrays for equality. Two arrays are equal if they have the same length
+     * and all characters at corresponding positions are equal.
+     *
+     * <p>Implementation Note: All characters in the first array are examined to determine equality.
+     * The calculation time depends only on the length of this first character array; it does not
+     * depend on the length of the second character array or the contents of either array.
+     *
+     * @param first the first array to compare
+     * @param second the second array to compare
+     * @return true if the arrays are equal, or false otherwise
+     */
+    public static boolean isEqual(char[] first, char[] second) {
+        if (first == second) {
+            return true;
+        }
+        if (first == null || second == null) {
+            return false;
+        }
+
+        if (second.length == 0) {
+            return first.length == 0;
+        }
+
+        // time-constant comparison that always compares all characters in first array
+        boolean matches = first.length == second.length;
```

Comment:
Can we return `false` directly if the lengths don't match? That would also remove the need for `int j = i < second.length ? i : 0;` when lengths match